Source/WTF/wtf/text/CodePointIterator.h

```diff
@@ -35,57 +35,57 @@ template<typename CharacterType>
 class CodePointIterator {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    ALWAYS_INLINE CodePointIterator() { }
-    ALWAYS_INLINE CodePointIterator(const CharacterType* begin, const CharacterType* end)
-        : m_begin(begin)
-        , m_end(end)
+    ALWAYS_INLINE CodePointIterator() = default;
+    ALWAYS_INLINE CodePointIterator(std::span<const CharacterType> data)
+        : m_data(data)
     {
     }
     
     ALWAYS_INLINE CodePointIterator(const CodePointIterator& begin, const CodePointIterator& end)
-        : CodePointIterator(begin.m_begin, end.m_begin)
+        : CodePointIterator({ begin.m_data.data(), end.m_data.data() })
     {
-        ASSERT(end.m_begin >= begin.m_begin);
+        ASSERT(end.m_data.data() >= begin.m_data.data());
     }
     
     ALWAYS_INLINE char32_t operator*() const;
     ALWAYS_INLINE CodePointIterator& operator++();
 
-    ALWAYS_INLINE friend bool operator==(const CodePointIterator&, const CodePointIterator&) = default;
+    ALWAYS_INLINE friend bool operator==(const CodePointIterator& a, const CodePointIterator& b)
+    {
+        return a.m_data.data() == b.m_data.data() && a.m_data.size() == b.m_data.size();
```

Comment:
Can’t we use the `std::span` version of `==` here? I’m surprised we can’t just use default like before.

Reply:
ahah, once would indeed assume that std::span has a comparison operator, and yet...
The standard currently doesn't have one, though I believe there is a proposal to add one.

Reply:
I see